### ISSUE TITLE
Always bumps with new patch versions unless passing with the argument?

### ISSUE THREAD

**Issue by qutang** [relekang/python-semantic-release#159]
It seems the tool always bumps with new patch versions, regardless of the types and number of changelogs it parses.

Will it possible to automatically detect what version level it should bump to based on the changelogs it parses.

For example, based on semantic versioning,
If there are incompatible API changes or BREAKING changes, it may bump with a new major version.
If there are feature updates, it may bump with a new minor version.
If there are only bug fixes, it may bump with a new patch version.

This is necessary for continuous integration because it is not always possible to pass the version level argument.

**Comment by relekang** [comment]
Can you give examples of config and git logs where this does not work? The whole point of this tool is to read this from the commit log and release the appropriate version. If it is behaving as you say that is a bug 🐛

**Comment by qutang** [comment]
I didn't config anything. The command I ran is just `semantic-release publish`.
The `semantic-release changelog` outputs multiple features, and bug fixes updates.

```
Feature
Support threads scheduler in pipeline (b13094b)
Add "max_processes" to the pipeline constructor to set the maximum processes to be used (cdfb092)
Pipeline class now works for multiple streams (bbe891e)
...
```

What git logs are you looking for?

I think this should be easy to reproduce by making some testing commits.

**Comment by relekang** [comment]
What happens when you run `DEBUG=semantic_release:* semantic-release publish`?

> I think this should be easy to reproduce by making some testing commits.

Yes, we have [tests for this](https://github.com/relekang/python-semantic-release/blob/master/tests/test_history.py#L38-L95) which are currently passing.

**Comment by relekang** [comment]
Closing this because of no response, please attach debug logs and reopen if this is still the case.

**Comment by ttimbers** [comment]
@relekang - wondering if we can re-open this issue, as I am facing a similar problem...

I am using `python-semantic-release` with the default commit parser on GitHub Actions. I am experiencing the same described issue. To quote @qutang above, "It seems the tool always bumps with new patch versions, regardless of the types and number of changelogs it parses."

I have tried various commit messages to no avail. Examples include a direct copy of a breaking change commit from the `angular.js` example (bottom of [this page](https://docs.google.com/document/d/1QrDFcIiPjSLDn3EL15IJygNPiHORgU1_OOAqWjiDU5Y/edit?usp=sharing)). This commit reads:

```
feat($compile): simplify isolate scope bindings

Changed the isolate scope binding options to:
  - @attr - attribute binding (including interpolation)
  - =model - by-directional model binding
  - &expr - expression execution binding

This change simplifies the terminology as well as
number of choices available to the developer. It
also supports local name aliasing from the parent.

BREAKING CHANGE: isolate scope bindings definition has changed and
the inject option for the directive controller injection was removed.

To migrate the code follow the example below:

Before:

scope: {
  myAttr: 'attribute',
  myBind: 'bind',
  myExpression: 'expression',
  myEval: 'evaluate',
  myAccessor: 'accessor'
}

After:

scope: {
  myAttr: '@',
  myBind: '@',
  myExpression: '&',
  // myEval - usually not useful, but in cases where the expression is assignable, you can use '='
  myAccessor: '=' // in directive's template change myAccessor() to myAccessor
}

The removed `inject` wasn't generaly useful for directives so there should be no code using it.
```

However it only resulted in a patch bump as observed here: 
<img width="985" alt="Screen Shot 2020-02-18 at 9 36 09 AM" src="https://user-images.githubusercontent.com/6865016/74762167-23c24c80-5232-11ea-964b-c9c3bf46f1a3.png">

I also tried using [one of your test commit messages](https://github.com/relekang/python-semantic-release/blob/ffc3f8bada5f9a031ffe3af1e00c01b0edb05740/tests/test_history.py#L12) that should result in a major version bump. 

This commit reads:
```
feat(x): Add super-feature


BREAKING CHANGE: Uses super-feature as default instead of dull-feature.
```
Same behaviour was observed, just a patch version bump was created:
<img width="996" alt="Screen Shot 2020-02-18 at 9 39 25 AM" src="https://user-images.githubusercontent.com/6865016/74762373-929fa580-5232-11ea-9ffb-511db4291d8e.png">

I also did as you suggest earlier, running it with `DEBUG=semantic_release:*` set as an env variable. Here's the output of the logs:

```
Run git config --local user.email "action@github.com"
From https://github.com/***/foocat
 * branch            master     -> FETCH_HEAD
Already up to date.
Creating new version..
Current version: 0.12.10
Bumping with a patch version to 0.12.11.
Tue, 18 Feb 2020 16:02:42 GMT semantic_release:cli 'main args:' {'force_level': None, 'post': False, 'retry': False, 'noop': False, 'define': ()} +0.0us
Tue, 18 Feb 2020 16:02:42 GMT semantic_release:cli 'main env:' 'PYPI_USERNAME="None",PYPI_PASSWORD="None",GH_TOKEN="$GH_TOKEN",GL_TOKEN="None",' +131.75us
Tue, 18 Feb 2020 16:02:42 GMT semantic_release:cli 'main config:' {'check_build_status': 'false', 'commit_message': 'Automatically generated by python-semantic-release', 'commit_parser': 'semantic_release.history.angular_parser', 'patch_without_tag': 'true', 'upload_to_pypi': 'false', 'version_source': 'commit'} +148.0us
Tue, 18 Feb 2020 16:02:42 GMT semantic_release:history 'get_current_version_by_config_file' +0.0us
Tue, 18 Feb 2020 16:02:42 GMT semantic_release:history 'foocat/__init__.py' '__version__' +56.25us
Tue, 18 Feb 2020 16:02:42 GMT semantic_release:history <re.Match object; span=(0, 23), match="__version__ = '0.12.10'"> +535.25us
Tue, 18 Feb 2020 16:02:42 GMT semantic_release:history:logs 'evaluate_version_bump("0.12.10", "None")' +0.0us
Tue, 18 Feb 2020 16:02:42 GMT semantic_release:vcs_helpers 'Reference v0.12.10 does not exist, considering all log' +0.0us
Tue, 18 Feb 2020 16:02:42 GMT semantic_release:history:parser_angular "parse_commit_message -> (2, feature, x, ('Add super-feature', '    BREAKING CHANGE: Uses super-feature as default instead of dull-feature. ', ''))" +0.0us
Tue, 18 Feb 2020 16:02:42 GMT semantic_release:history 'get_new_version("0.12.10", "patch")' +11.1ms
Bumping version from 0.12.10 to 0.12.11
[master 92881de] Add changes
 1 file changed, 1 insertion(+), 1 deletion(-)
```

If it's helpful, here is my GitHub Actions workflow: https://github.com/ttimbers/foocat/blob/master/.github/workflows/release.yml

*note - I opened the #175 because I need to bump both an `__init__.py` file and `pyproject.toml`, and my work around currently is to use `poetry run semantic-release version` followed by `poetry version ...` and then manually committing and pushing changes to bump both files.*

And my commit history for the repo I am testing this out on: https://github.com/ttimbers/foocat/commits/master

**Comment by ttimbers** [comment]
Thanks for re-opening @relekang !

**Comment by relekang** [comment]
@ttimbers Can you try 4.6.0?

**Comment by ttimbers** [comment]
@relekang - I have upgraded my dep's to 4.6.0 but it still doesn't seem to be working on my CI. I tried 3 times with 3 different commit messages... Here is the log from the last commit message (which I also paste below) and the version bump it did was just from  0.12.22 to 0.12.23.

Commit message:
```
feat(x): Add super-feature


BREAKING CHANGE: Uses super-feature as default instead of dull-feature.
```

CI logs:
```
Run git config --local user.email "action@github.com"
From https://github.com/***/foocat
 * branch            master     -> FETCH_HEAD
Already up to date.
Creating new version..
Current version: 0.12.22
Bumping with a patch version to 0.12.23.
Wed, 19 Feb 2020 20:24:15 GMT semantic_release:cli 'main args:' {'force_level': None, 'post': False, 'retry': False, 'noop': False, 'define': ()} +0.0us
Wed, 19 Feb 2020 20:24:15 GMT semantic_release:cli 'main env:' 'PYPI_USERNAME="None",PYPI_PASSWORD="None",GH_TOKEN="$GH_TOKEN",GL_TOKEN="None",' +72.75us
Wed, 19 Feb 2020 20:24:15 GMT semantic_release:cli 'main config:' {'check_build_status': 'false', 'commit_message': 'Automatically generated by python-semantic-release', 'commit_parser': 'semantic_release.history.angular_parser', 'patch_without_tag': 'true', 'upload_to_pypi': 'false', 'version_source': 'commit'} +82.25us
Wed, 19 Feb 2020 20:24:15 GMT semantic_release:history 'get_current_version_by_config_file' +0.0us
Wed, 19 Feb 2020 20:24:15 GMT semantic_release:history 'foocat/__init__.py' '__version__' +30.0us
Wed, 19 Feb 2020 20:24:15 GMT semantic_release:history <re.Match object; span=(0, 23), match="__version__ = '0.12.22'"> +394.25us
Wed, 19 Feb 2020 20:24:15 GMT semantic_release:history:logs 'evaluate_version_bump("0.12.22", "None")' +0.0us
Wed, 19 Feb 2020 20:24:15 GMT semantic_release:vcs_helpers 'Reference v0.12.22 does not exist, considering all log' +0.0us
Wed, 19 Feb 2020 20:24:15 GMT semantic_release:history:logs 'ignored commit' UnknownCommitMessageStyleError("Unable to parse the given commit message: Merge branch 'master' of github.com:***/foocategoricals\n") +10.2ms
Wed, 19 Feb 2020 20:24:15 GMT semantic_release:history:logs 'commit_count=1' +189.75us
Wed, 19 Feb 2020 20:24:15 GMT semantic_release:history:logs 'Changing bump to patch based on config patch_without_tag' +57.0us
Wed, 19 Feb 2020 20:24:15 GMT semantic_release:history:logs 'evaluate_version_bump returned patch' +46.25us
Wed, 19 Feb 2020 20:24:15 GMT semantic_release:history 'get_new_version("0.12.22", "patch")' +10.5ms
Bumping version from 0.12.22 to 0.12.23
[master 212c20a] Add changes
 1 file changed, 1 insertion(+), 1 deletion(-)
```

**Comment by ttimbers** [comment]
This line might be where the error is occurring:

```
Wed, 19 Feb 2020 20:24:15 GMT semantic_release:history:logs 'ignored commit' UnknownCommitMessageStyleError("Unable to parse the given commit message: Merge branch 'master' of github.com:***/foocategoricals\n") +10.2ms
```

However, that doesn't quite make sense to me given that I copied the commit message from your tests?

**Comment by relekang** [comment]
I checked the github action log and it clones the repo with depth=1 so the only available commit is the merge commit.

**Comment by ttimbers** [comment]
OK, thanks for tracking down my error - I am new to GitHub Actions, and I missed this. I'll look into the docs and see how I can clone with more commits (i.e., depth > 1) and report back.

**Comment by ttimbers** [comment]
This does indeed fix it. I apologize for sending you down a 🐇 hole. Closing this issue now, and many thanks for your help. This is a very cool project and I am grateful for your work on it.

**Comment by relekang** [comment]
No worries, we found a pretty bad bug and got it fixed with 1daa4e2. Thank you for the thorough description, it helped a lot.
